### PR TITLE
Support Inference with CAL Model

### PR DESCRIPTION
**Pull request by darkushin**
Add CAL inference to our pipeline

### CODE REVIEW

File: models/model_runner.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 from DataProcessing.DB.dal import Crop, get_entries
 from DataProcessing.utils import viz_DB_data_on_video
-from models.model_constants import *
```
Comment (bar371) [inline models/model_runner.py:10]: Does it catch model_constants now? it is not under 'models'?

File: models/track_and_reid_model.py [hunk 2]
```diff
@@ -455,6 +456,18 @@ def create_data_by_re_id_and_track():
...
+
+        # initialize reid model:
+        reid_model = CAL_build_model(reid_cfg)
+
+        # create gallery features:
+        gallery_data = CAL_make_inference_data_loader(reid_cfg, reid_cfg.DATA.ROOT, ImageDataset)
+        g_feats, g_paths = CAL_run_inference(reid_model, gallery_data, args.device)
+        g_pids = np.array([pid.split('/')[-1].split('_')[0] for pid in g_paths]).astype(int)  # need to be only the string id of a person ('0015' etc.)
+        g_feats = torch.from_numpy(g_feats)
     else:
```
Comment (bar371) [inline models/track_and_reid_model.py]: Would switch this to 'CTL' and on else throw an exception that the ReID model selected is now supported

File: models/track_and_reid_model.py [hunk 3]
```diff
@@ -455,6 +456,18 @@ def create_data_by_re_id_and_track():
...
         feats, g_feats, g_pids, g_camids = load_reid_features(output_path=FAST_PICKLES)
+    elif args.reid_model == 'CAL':
+        reid_cfg = set_CAL_reid_cfgs(args)
+        assert len(os.listdir(reid_cfg.DATA.ROOT)) > 0, "reid gallery is empty, check args."
+
+        # initialize reid model:
+        reid_model = CAL_build_model(reid_cfg)
+
+        # create gallery features:
+        gallery_data = CAL_make_inference_data_loader(reid_cfg, reid_cfg.DATA.ROOT, ImageDataset)
```
Comment (bar371) [inline models/track_and_reid_model.py:502]: question - does the DATA.ROOT need to be in fast-reid format? or do we just give the exact path (bounding_box_test?)

File: models/CAL_reid_inference.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    features = torch.cat(features, 0)
+
+    return features, pids, camids
+
+
+def build_img_transforms(config):
+    transform_test = T.Compose([
+        T.Resize((config.DATA.HEIGHT, config.DATA.WIDTH)),
+        T.ToTensor(),
+        T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
```
Comment (bar371) [inline models/CAL_reid_inference.py:132]: Makes me wonder regarding the tensors we are using in CAL, are they normalised now as well?

File: models/model_runner.py (hunk 1 above)
Comment (darkushin) [reply models/model_runner.py:10]: This didn't work for me when it was with `models.`. 
I suggest we test it from you before we merge.

File: models/track_and_reid_model.py (hunk 3 above)
Comment (darkushin) [reply models/track_and_reid_model.py:502]: This is the full path (including bounding_box_test).

File: models/CAL_reid_inference.py (hunk 4 above)
Comment (darkushin) [reply models/CAL_reid_inference.py:132]: You mean CTL right?
We call something that normalized both the gallery and a given track using this
image

Comment (bar371) [reply models/CAL_reid_inference.py:132]: Great! tnx

File: models/model_runner.py (hunk 1 above)
Comment (bar371) [reply models/model_runner.py:10]: I had the same issue on Master --- it could be solved by marking collect root paths on pycharm
image

**Review by bar371 — APPROVED** [review]
